src/modules/ipfix/aggregator/AggregatorBaseCfg.cpp

```diff
@@ -94,14 +94,15 @@ Rule* AggregatorBaseCfg::readRule(XMLElement* elem) {
 		delete rule;
 		rule = NULL;
 	}
-
-	// exclude coexistence of patterns and biflow aggregation
-	if(rule->biflowAggregation) {
-		for(int i=0; i < rule->fieldCount; i++) {
-			if(rule->field[i]->pattern)
-				msg(MSG_ERROR, "AggregatorBaseCfg: Match pattern for id=%d ignored because biflow aggregation is enabled.", rule->field[i]->type.id);
-				free(rule->field[i]->pattern);
-				rule->field[i]->pattern = NULL;
+	else {
+		// exclude coexistence of patterns and biflow aggregation
+		if(rule->biflowAggregation) {
+			for(int i=0; i < rule->fieldCount; i++) {
+				if(rule->field[i]->pattern)
+					msg(MSG_ERROR, "AggregatorBaseCfg: Match pattern for id=%d ignored because biflow aggregation is enabled.", rule->field[i]->type.id);
+					free(rule->field[i]->pattern);
```

Comment:
Indentation does not match -> Add braces to make clear what is part of the if body.

Reply:
I think the original code was wrong here, and the free should only be in the then branch

Reply:
As I interpret the comment, it says that biflow aggregation in combination with pattern matching is not supported. Therefore, if we find a pattern with biflow aggreagation, we throw the error, free() and set to NULL. What do you think @evintila?

Reply:
I agree